core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineSqlServer.java

```diff
@@ -223,14 +223,16 @@ public DataType getDataTypeForScalarDb(
         return DataType.TEXT;
       case BINARY:
       case VARBINARY:
-        if (typeName.equalsIgnoreCase("timestamp") || typeName.equalsIgnoreCase("hierarchyid")) {
+        if (!typeName.equalsIgnoreCase("binary") && !typeName.equalsIgnoreCase("varbinary")) {
           throw new IllegalArgumentException(
               String.format("Data type %s is unsupported: %s", typeName, columnDescription));
         }
-        logger.info(
-            "Data type larger than that of underlying database is assigned: {} ({} to BLOB)",
-            columnDescription,
-            typeName);
+        if (columnSize < Integer.MAX_VALUE) {
```

Comment:
I missed that SQL server users can specify `varbinary(max)` for the binary data larger than 8000 bytes (`varbinary(8000)`), and it can hold the same size as ScalarDB BLOB. In that case, the message is unnecessary.